wwwapp/views.py

```diff
@@ -379,6 +380,51 @@ def participants_view(request, year):
 
     return render(request, 'participants.html', context)
 
+@login_required()
+@permission_required('wwwapp.see_all_workshops', raise_exception=True)
+def lecturers_view(request: HttpRequest, year: int) -> HttpResponse:
+    year = int(year)
+
+    workshops = Workshop.objects.filter(type__year=year).prefetch_related('lecturer', 'lecturer__user')
+
+    people: Dict[int, Dict[str, any]] = {}
+    for workshop in workshops:
+        for lecturer in workshop.lecturer.all():
+            lecturer_id = lecturer.id
+
+            if lecturer_id in people:
+                people[lecturer_id]['workshops'].append(workshop.info_for_client_link())
```

Comment:
Why not append the whole workshop object here?

Reply:
We could remove `info_for_client_link()` method then.

Reply:
I don't want to pass unnecessary data to the template.

Reply:
I don't see a reason not to, but whatever.